homeassistant/config.py

```diff
@@ -824,18 +842,36 @@ async def merge_packages_config(
 
 
 async def async_process_component_config(  # noqa: C901
-    hass: HomeAssistant, config: ConfigType, integration: Integration
+    hass: HomeAssistant,
+    config: ConfigType,
+    integration: Integration,
+    *,
+    raise_on_failure: bool = False,
 ) -> ConfigType | None:
     """Check component configuration and return processed configuration.
 
     Returns None on error.
+    Optional an exception is raised if the configuration is not valid,
+    but returns None on other errors.
```

Comment:
On which errors does it return None even if `raise_on_failure` is `True`?

Reply:
It used to be after previous commits.